rust/worker/src/execution/orchestration/compact.rs

```diff
@@ -309,7 +310,7 @@ impl CompactOrchestrator {
             },
         };
 
-        self.num_write_tasks = partitions.len() as i32;
+        self.num_write_tasks = partitions.len() as i32 * 3; // 3 different segment types
```

Comment:
This seems worth calling out as an assumption for when we add the index.  I don't know the best way to log it.  I'd do a `TODO(feat)`, but we need to agree on what that would mean.